### PR TITLE
[EV-5299] Disable Policy Recommendation for Windows OS

### PR DESCRIPTION
**Pull request by dimitri-nicolo**
## Description

Disable Image Resolution for Windows
* Policy Recommendations is unsupported until Staged Network Policies are supported in Windows

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/policyrecommendation/policyrecommendation_controller.go
Comment (rene-dekker) [inline pkg/controller/policyrecommendation/policyrecommendation_controller.go:434]: If I understand it correctly, we cannot have a cluster using policy recommendation and windows nodes at the same time, because windows does not support staged policies.

If that is the case, we should degrade the TigeraStatus immediately if hasWindowsNodes is true. I think we can move this if-statement further up, so we may save some unnecessary calculations, perhaps right underneath the ready checks at line 291.

Comment (dimitri-nicolo) [reply pkg/controller/policyrecommendation/policyrecommendation_controller.go]: Moved the check to below line 291.

It isn't used, but I've kept the check in the render to enforce a defensive pattern

File: pkg/controller/policyrecommendation/policyrecommendation_controller.go
Comment (rene-dekker) [inline pkg/controller/policyrecommendation/policyrecommendation_controller.go:298]: I was wondering if we could make it a bit more descriptive, so they understand well what's going on.

> Policy Recommendation is not supported in combination with Windows nodes.

File: pkg/controller/policyrecommendation/policyrecommendation_controller.go
Comment (rene-dekker) [inline pkg/controller/policyrecommendation/policyrecommendation_controller.go:294]: Judging by the implementation of HashWindowsNodes, I think it should be a `ResourceReadError` and perhaps the following message would be better.

> Unable to determine if there are windows nodes.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Pull request merged**
